File: Chroma-Core/src/main/java/buttondevteam/lib/ScheduledServerRestartEvent.java
package buttondevteam.lib;

import buttondevteam.core.component.restart.ScheduledRestartCommand;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

@Getter
@RequiredArgsConstructor
public class ScheduledServerRestartEvent extends Event {
	private static final HandlerList handlers = new HandlerList();

	private final int restartTicks;
	private final ScheduledRestartCommand command;

	@Override
	public HandlerList getHandlers() {
		return handlers;
	}

	public static HandlerList getHandlerList() {
		return handlers;
	}
}

File: Chroma-Core/src/main/java/buttondevteam/lib/ScheduledServerRestartEvent.kt
package buttondevteam.lib

import buttondevteam.core.component.restart.ScheduledRestartCommand
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class ScheduledServerRestartEvent(val restartTicks: Int, val command: ScheduledRestartCommand) : Event() {
    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        @JvmStatic
        val handlerList = HandlerList()
    }
}